### DISCUSSION TITLE
5.2 Release Planning

### DISCUSSION THREAD

**Discussion by jonmmease** [vega/altair#3271] [General]
Hey All,

I'd like to start talking about when we want to do a 5.2 release, and what else we want to have included. I'm interested in getting the PDF, toUrl, and offline HTML support released relatively soon.

At minimum, I want to merge this JupyterChart fix (https://github.com/altair-viz/altair/pull/3269), and it would be great to also update Vega to 5.26.1 and Vega-Lite to 5.16.3.

What else would folks like to have included?

**Comment by joelostblom** [comment]
Those all sounds like great fixes/upgrades to have in 5.2! I don't think I have anything particular in mind atm. Unfortunately, I have minimum capacity to help until mid December, but I will try to be as responsive as I can in case there is anything I need to weigh in on.

**Comment by binste** [comment]
Thanks for kicking this off! In addition to the already mentioned ones, it
would be great if we can finish the type hinting of Altair and mark it as a
typed package. For this, it would be great if someone could do another
review on
https://github.com/altair-viz/altair/pull/3208 (Mattijn already reviewed
and approved) and I'll then create a small follow-up PR which should be a
quick thing.

I'd suggest that we tackle the VL upgrade after the type hint as they
modify the same files.

Looking forward to another feature release this year :)

**Comment by jonmmease** [comment]
Getting the type hint work into 5.2 would be great! I just tested and reviewed #3208 and I think it's pretty much ready to go.  Is there anything else left before we can mark Altair as a typed package?

I'm interested to try my hand at making the Vega and Vega-Lite updates this time around if no one else was looking forward to it.

**Reply by binste** [reply]
Thanks for the review! I now created a smaller follow-up PR which cleans up some loose ends and marks Altair as a typed package: https://github.com/altair-viz/altair/pull/3272

Sure, happy to review the update PR and just ping me if you're stuck somewhere :) In case you haven't seen it yet, the upgrade instructions are in `NOTES_FOR_MAINTAINERS.md`.

**Comment by ChristopherDavisUCI** [comment]
All sounds great to me, thanks for these updates!  I haven't had much time to devote to Altair the past few months, and when I've found time I've mostly been thinking about the Vega-Lite side, especially related to violin plots related to @joelostblom's comment [here](https://github.com/vega/vega-lite/issues/3442#issuecomment-1764465275).

**Comment by jonmmease** [comment]
Oh, and I'd also like to resolve https://github.com/altair-viz/altair/issues/3261 by propagating locale info through to vl-convert when saving to static images.

**Comment by jonmmease** [comment]
I think everything mentioned in this thread has been merged. I believe @mattijn and @joelostblom are the only ones who have permissions to perform a release (is that correct?). Since @joelostblom isn't available for a few weeks, @mattijn would you be available to make a 5.2.0 release?

Does anyone have opinions on whether an RC is warranted?

**Reply by mattijn** [reply]
I can cut a release. Are there notable regressions that we are willing to accept? Eg. https://github.com/vega/vega-lite/issues/9143?

Making Altair a typed library is the biggest change, I can't oversee to what end this requires a release candidate. What you think @binste?

**Reply by mattijn** [reply]
Also, I noticed that the colors of pydata sphinx extension changed, especially the second-primary color. 
Edit: fixed by https://github.com/altair-viz/altair/pull/3276

**Reply by jonmmease** [reply]
Considering regressions in Vega-Lite is a valid consideration.  The Vega-Lite update was pretty minor from the Altair API side, so we could consider reverting it for 5.2. Are there other regressions you're tracking @mattijn?

**Reply by mattijn** [reply]
No I'm not aware of more regressions (I think https://github.com/vega/vega-lite/issues/9142 was fixed already by Vega [v5.26.1](https://github.com/vega/vega/releases/tag/v5.26.1)), maybe https://github.com/vega/vega-lite/issues/9143 is even fixed by this bump of Vega as well?

**Reply by mattijn** [reply]
I cannot reproduce https://github.com/vega/vega-lite/issues/9142 and https://github.com/vega/vega-lite/issues/9143 on our main branch. Let me proceed with making a 5.2 release.

**Reply by binste** [reply]
I think it would be ok to not do a release candidate. If there are issues with the type hints, it shouldn't break users code and mypy errors in a CICD pipeline (or wherever it runs) can be individually disabled by the users themselves until we'd release a fix. Thanks @mattijn for doing the release! 🥳

**Reply by mattijn** [reply]
Done per https://github.com/altair-viz/altair/releases/tag/v5.2.0.

**Reply by binste** [reply]
Awesome, thanks! (I marked the "Release" on GitHub as "latest")
